lib/secure-boot-lib.robot

```diff
@@ -0,0 +1,213 @@
+    @{menu_construction}=    Create List
+    FOR    ${line}    IN    @{menu_lines}
+        ${line}=    Remove String    ${line}    -    \\    \    /    |    <    >
+        ${line}=    Replace String Using Regexp    ${line}    ${SPACE}+    ${SPACE}
+        IF    "${line}"!="${EMPTY}" and "${line}"!=" "
+            ${line}=    Strip String    ${line}
+            Append To List    ${menu_construction}    ${line}
+        END
+    END
+    ${menu_construction}=    Get Slice From List    ${menu_construction}[1:]
+    RETURN    ${menu_construction}
+
+Upload Image
+    [Documentation]    Mounts the image from the given URL on the PiKVM.
+    [Arguments]    ${ip}    ${img_url}
+    Upload Image To PiKVM    ${ip}    ${img_url}
+
+Mount Image
+    [Documentation]    Mounts the image with the given name on the PiKVM.
+    [Arguments]    ${ip}    ${img_name}
+    Mount Image On PiKVM    ${ip}    ${img_name}
+
+Get File Explorer Submenu Construction
+    [Documentation]    Keyword allows to get and return File Explorer menu construction.
+    # Read From Terminal Until    NEW FILE
+    ${menu}=    Read From Terminal Until    Move Highlight
+    @{menu_lines}=    Split To Lines    ${menu}
+    @{menu_lines}=    Remove Entry From List    ${menu_lines}    .*Move Highlight.*
+    @{menu_construction}=    Create List
+    FOR    ${line}    IN    @{menu_lines}
+        ${line}=    Remove String    ${line}    -    \\    \    /    |    <    >
+        ${line}=    Replace String Using Regexp    ${line}    ${SPACE}+    ${SPACE}
+        IF    "${line}"!="${EMPTY}" and "${line}"!=" "
+            ${line}=    Strip String    ${line}
+            Append To List    ${menu_construction}    ${line}
+        END
+    END
+    ${menu_construction}=    Get Slice From List    ${menu_construction}[1:]
+    RETURN    ${menu_construction}
+
+Enter UEFI Shell And Boot .EFI File
+    [Documentation]    Boots given .efi file from UEFI shell.
+    ...    Assumes that it is located on FS0.
+    [Arguments]    @{filename}
+    Enter Boot Menu Tianocore
+    Enter UEFI Shell Tianocore
+    Read From Terminal Until    Shell>
+    Boot .EFI File From UEFI Shell    @{filename}
+
+Boot .EFI File From UEFI Shell
+    [Documentation]    Boots given efi file in UEFI shell using PiKVM,
+    ...    and checks the result
+    [Arguments]    ${filename}    ${expected_result}
+    Sleep    3s
+    Write Bare Into Terminal    fs0:
+    Write Bare Into Terminal    \n
+    Sleep    1s
+    Write Bare Into Terminal    ${filename}
+    Write Bare Into Terminal    \n
+    Read From Terminal Until    ${expected_result}
+
+Verify Reset Secure Boot Keys
+    [Documentation]    Verifies that Reset Secure Boot Keys
+    ...    option is available.
+    Enter Setup Menu Tianocore
+    Enter Device Manager Submenu
+    Enter Secure Boot Configuration Submenu
+    Read From Terminal Until    Reset Secure Boot Keys
+
+Reset Secure Boot Keys Again
```

Comment:
Why do we need this strange `Reset Secure Boot Keys **Again**` KW? We simply need KW to reset keys, that's it.
Like I asked to just copy from: https://github.com/Dasharo/open-source-firmware-validation/commit/7e69d7ec371ec73d1276830bc0108cb862b71cb4#diff-b06ff90d8f021dfa2e18d217f3d4c1e6861d6f671893cf292b6718b80227921aR71

Reply:
Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)